rt/transports/jms/src/main/java/org/apache/cxf/transport/jms/JMSConduit.java

```diff
@@ -282,6 +283,9 @@ private void sendAndReceiveMessage(final Exchange exchange, final Object request
                                                                      jmsConfig.getReceiveTimeout(),
                                                                      jmsConfig.isPubSubNoLocal(),
                                                                      exchange);
+                    if (replyDestination instanceof TemporaryQueue) {
```

Comment:
Thanks for the patch @valentin-matignon , I believe the this is not the right place to delete the temporary queue. 
 - `JMSConfiguration` is the class that create temporary queue (and caches it)
 - `JMSConfiguration` has the method `resetCachedReplyDestination` which looks like the right place to delete the temporary queue (it caches the destination in `replyToDestinationDest`)
 - the `JMSConduit` should be likely calling  `JMSConfiguration::resetCachedReplyDestination()` on shutdown
 - it would be great to have at least one test case for the issue

What do you think? Thank you.

Reply:
Hi @reta, thank you for your answer and sorry to respond so late !

I agree with you, `JMSConfiguration` should delete temporary queues.
As you advised, in my last commit i delete temporary queues in the `resetCachedReplyDestination` method of the `JMSConfiguration` class.

However, `resetCachedReplyDestination` is called in two places places in the `JMSConduit` class :

- in a `ExceptionListener` in the `trySetExListener` method
- in the `sendExchange` method when a `JMSException` occurs

So i had to add the `resetCachedReplyDestination` in the `close` method of `JMSConduit` class to be sure that temporary queues are deleted every time they are used.

I didn't find a proper way to add one test case in unit tests, if you have some suggestions i could add it in my PR.